### ISSUE TITLE
Common structure to define schema elements in prose

### ISSUE THREAD

**Issue by tlimmer** [oasis-tcs/csaf#158]
At the moment we have unstructured prose text that describes both the semantics and syntax of the CSAF schema elements. This is very hard to read and understand. 
I propose to introduce a common structure to the prose text. The first sentence(s) after the heading introduce the semantics of the element, then we include a bulleted list that gives more information about the JSON structure.

**Short example:**
Original prose:

> #### 3.1.1.1 Acknowledgments Type - Acknowledgment
> The value type of `Acknowledgment` is object with at least 1 and at most 4 properties. Every such element acknowledges contributions by describing those that contributed.
> The properties are: `names`, `organizations`, `summary`, and `urls`.

New prose:

> #### 3.1.1.1 Acknowledgments Type - Acknowledgment
> Acknowledges contributions by describing those that contributed.
> 
> * Defined Element: `Acknowledgment`
> * Element Type: `object` (between 1 and 4 properties)
> * Allowed properties: `names`, `organizations`, `summary`, `urls`

**Longer example (without original prose):**

> ### 3.1.1 Acknowledgments Type
> Contains a list of acknowledgments that describe contributors of the document.
> 
> * Defined Element: `acknowledgments_t` 
> * Element Type: array (1 or more items)
> * Array Item Type: `Acknowledgement`
> 
> **Schema excerpt:**
> 
>     "acknowledgments_t": {
>       // ...
>       "items": {
>         "title": "Acknowledgment"
>         "properties": {
>           "names": {
>             // ...
>           },
>           "organizations": {
>             // ...
>           },
>           "summary": {
>             // ...
>           },
>           "urls": {
>             // ...
>           }
>         }
>       }
>     },
> 
> #### 3.1.1.1 Acknowledgments Type - Acknowledgment
> Acknowledges contributions by describing those that contributed.
> 
> * Defined Element: `Acknowlegment`
> * Element Type: `object` (between 1 and 4 properties)
> * Allowed properties: `names`, `organizations`, `summary`, `urls`
> 
> ##### 3.1.1.1.1 Acknowledgments Type - Acknowledgment - Names
> Holds the names of contributing persons being recognized. Each list item describes a single person.
> 
> * Defined Element: `names`
> * Element Type: `array` (1 or more items)
> * Array Item Type: `string` (1 or more characters)
> 
> **Example:**
>     
>     "names": [
>       "Johann Sebastian Bach",
>       "Albert Einstein"
>     ]
> 
> ##### 3.1.1.1.2 Acknowledgments Type - Acknowledgment - Organizations
> Holds the names of contributing organizations being recognized. Each list item describes a single organization.
> 
> * Defined Element: `organizations`
> * Element Type: `array` (1 or more items)
> * Array Item Type: `string` (1 or more characters)
> 
> **Example:**
> 
>     "organizations": [
>       "CISA",
>       "Talos",
>       "Google Project Zero"
>     ]

**Comment by tschmidtb51** [comment]
We also should discuss whether we introduce headings for "anonymous object" (see above "Acknowledgement"). I prefer that but it isn't done yet throughout the prose.

**Comment by tlimmer** [comment]
> We also should discuss whether we introduce headings for "anonymous object" (see above "Acknowledgement")

What anonymous object do you mean here?

**Comment by tschmidtb51** [comment]
> > We also should discuss whether we introduce headings for "anonymous object" (see above "Acknowledgement")
> 
> What anonymous object do you mean here?

The items within the `acknowledgments_t`:
```
    "items": {
      "title": "Acknowledgment"
      "properties": {
        "names": {
          // ...
        },
        "organizations": {
          // ...
        },
        "summary": {
          // ...
        },
        "urls": {
          // ...
        }
      }
    }
```
They have a `title` but no real name.

**Comment by sthagen** [comment]
> This is very hard to read and understand.
I do not share this perspective and perceive the list structure as competing with the original json schema presentation.
Requirements linkage is hard in English prose but even harder across a third perspective via these pseudo grammar list segmentation.

We already have a common approach for the prose describing the elements and we may spot deviations and correct them.

I suggest to avoid introducing a pseudo json schema presentation language and then fight with conformance formulation.

But, I will investigate and can maybe present such an alternative formulation to the TC members in one of the next two meetings.

**Comment by tschmidtb51** [comment]
> Requirements linkage is hard in English prose but even harder across a third perspective via these pseudo grammar list segmentation.
> 
> We already have a common approach for the prose describing the elements and we may spot deviations and correct them.
> 
> I suggest to avoid introducing a pseudo json schema presentation language and then fight with conformance formulation.

I agree with @sthagen. We already have a description which a programmer can use. The prose should be structured but still complete sentences. If we stick to a form with those, it also allows the reader to get used to it and find the information quickly.

**Comment by tlimmer** [comment]
> I do not share this perspective and perceive the list structure as competing with the original json schema presentation.
> Requirements linkage is hard in English prose but even harder across a third perspective via these pseudo grammar list segmentation.

Yes, I understand this aspect. My original reason to propose the structured lists was, that it was harder for me to understand the prose sentences than the actual JSON schema definition. Maybe we can find some kind of compromise here, as mainly programmers will need to read the spec?

**Comment by sthagen** [comment]
I propose  to close after inspection of the upcoming editor draft. I perceive that the readability has become much better and we might have solved this issue en passant.

**Comment by tschmidtb51** [comment]
Closed on request of @tlimmer.
